docs/content/docs/recipes/signals.md

```diff
@@ -0,0 +1,328 @@
+
+The class defines the `focusSignalType` that will be sent and listened to the connected clients when there is a presence change:
+
+```typescript
+export class FocusTracker extends TypedEventEmitter<IFocusTrackerEvents> {
+    private static readonly focusSignalType = "changedFocus";
+    /*...*/
+}
+```
+The class then initializes a local map of boolean values for all of the connected clients. The boolean denotes whether or not the client has focus or not:
+
+```typescript
+private readonly focusMap = new Map<string, Map<string, boolean>>();
+```
+We can now move to the constructor where we can see that `FocusTracker` takes in the `container`, the `audience`, and the `Signaler` instance from `initialObjects` as arguments:
+
+```typescript
+constructor(
+  container: IFluidContainer,
+  public readonly audience: IServiceAudience<IMember>,
+  private readonly signaler: Signaler,
+) {
+  super();
+
+  /*...*/
+}
+```
+Just like in the `MouseTracker` class, whenever there is any change to a client's focus status we'll fire a `focusChanged` event to notify listeners that updated data is available:
+
+```typescript
+export interface IFocusTrackerEvents extends IEvent {
+    (event: "focusChanged", listener: () => void): void;
+}
+```
+Also similar to `MouseTracker`, a member leaves the audience we need to remove their presence from our local data and emit an event to let the view know that it needs to re-render to display the updated presence:
+
+```typescript
+this.audience.on("memberRemoved", (clientId: string, member: IMember) => {
+    const focusClientIdMap = this.focusMap.get(member.userId);
+    if (focusClientIdMap !== undefined) {
+        focusClientIdMap.delete(clientId);
+        if (focusClientIdMap.size === 0) {
+            this.focusMap.delete(member.userId);
+        }
+    }
+    this.emit("focusChanged");
+});
+```
+To track the local client's focus status, we use the `focus` and `blur` events to know when the focus boolean must be updated. To alert the connected clients of this focus change, we then submit a `focusSignalType` signal with the client's userID and updated focus status as the paylod:
+
+```typescript
+window.addEventListener("focus", () => {
+    this.sendFocusSignal(true);
+});
+window.addEventListener("blur", () => {
+    this.sendFocusSignal(false);
+});
+```
+```typescript
+private sendFocusSignal(hasFocus: boolean) {
+    this.signaler.submitSignal(
+        FocusTracker.focusSignalType,
+        { userId: this.audience.getMyself()?.userId, focus: hasFocus },
+    );
+}
+```
+To make sure all connected clients are notified of this focus change, we use the `onSignal` function to listen to the `focusSignalType`. We then update the local data using the payload information from the signal and emit a `focusChanged` event to re-render:
+
+```typescript
+this.signaler.onSignal(FocusTracker.focusSignalType, (clientId, local, payload) => {
```

Comment:
The corresponding line in the MouseTracker above specified data types:

```
(clientId: string, local: boolean, payload: Jsonable) 
```

Is there a reason for the difference? I think specifying the types makes the code more readable/understandable, although the actual example app that is linked to at the top of this article doesn't specify the types. (Maybe it should?)